File: src/NuFridge.Website.MVC/Models/NuFridgeInstall.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace NuFridge.Website.MVC.Models
{
    public class NuFridgeInstall
    {
        public int PortNumber { get; set; }
        public string MongoDBServer {get;set;}
        public string MongoDBDatabase { get; set; }
        public string IISWebsiteName { get; set; }
        public string PhysicalDirectory { get; set; }
    }
}

File: src/NuFridge.Website.MVC/App/viewmodels/databinding/NuFridgeInstall.js
define(['knockoutvalidation'], function (validation) {

    ko.validation.init({
        registerExtenders: true,
        insertMessages: false,
    });

    window.NuFridgeInstall = function (config) {
        var self = this, data;

        data = $.extend({
            MongoDBServer: ko.observable("").extend({ required: true }),
            MongoDBDatabase: ko.observable("").extend({ required: true }),
            IISWebsiteName: ko.observable("").extend({ required: true }),
            PortNumber: ko.observable("80").extend({
                required: true,
                pattern: { message: 'Port number must be numeric', params: /^\d+$/ }
            }),
            PhysicalDirectory: ko.observable("").extend({
                required: true,
                pattern: { message: 'Please provide a valid directory path', params: /^([A-Za-z]:)(\\[A-Za-z_\-\s0-9\.]+)+$/ }
            })
        }, config);

        ko.mapping.fromJS(data, {}, self);
    };

    window.NuFridgeInstall.mapping = {
        create: function (options) {

            var fd = new NuFridgeInstall(options.data);

            return fd;
        }

    };
});